### ISSUE TITLE
[Bug]: DC0005 `XmlDocumentationProcedureConsistency` throws unhandled `InvalidOperationException` instead of reporting a diagnostic

### ISSUE THREAD

**Issue by finnpedersenkazes** [ALCops/Analyzers#417]
### Analyzer

DocumentationCop

### Issue type

Crash / exception

### Rule ID

DC0005

### Description

**Warning**: All of the following is AI generated based on some errors I experienced when migrating to ALCops. 
Since I am not a `C#` developer, I am unable to do the development and a PR. However, the AI have suggested a solution. 

> I hope that the issue contains all needed information, so you can assign the issue directly to the AI.

## Description

The `ALCops.DocumentationCop.Analyzers.XmlDocumentationProcedureConsistency` analyzer crashes with an unhandled `System.InvalidOperationException` ("Sequence contains no matching element") instead of gracefully emitting a diagnostic warning/error.

The crash is triggered by either of the following malformed XML documentation patterns:

1. A `<param>` tag with **no `name` attribute** (e.g. `<param></param>`) on a procedure that has parameters.
2. A `<param>` tag with **no `name` attribute** on a procedure that has **no parameters at all**.
3. *(Possibly)* A `<param name="X">` where `X` does not match any actual parameter in the procedure signature.

In all cases the analyzer should emit a diagnostic (a warning or error) rather than throw.

**Expected behaviour:** The analyzer emits a `DC0005` diagnostic pointing at the malformed `<param>` tag.

**Actual behaviour:** The analyzer throws `System.InvalidOperationException: Sequence contains no matching element` and no diagnostic is produced.

## Exception

```
Analyzer 'ALCops.DocumentationCop.Analyzers.XmlDocumentationProcedureConsistency'
threw an exception of type 'System.InvalidOperationException' with message
'System.InvalidOperationException: Sequence contains no matching element
   at System.Linq.ThrowHelper.ThrowNoMatchException()
   at ALCops.DocumentationCop.Analyzers.XmlDocumentationProcedureConsistency
         .AnalyzeDocumentationComments(SyntaxNodeAnalysisContext ctx)
   at Microsoft.Dynamics.Nav.CodeAnalysis.Diagnostics.AnalyzerExecutor
         .<>c__DisplayClass53_1.<ExecuteSyntaxNodeAction>b__1()
   at Microsoft.Dynamics.Nav.CodeAnalysis.Diagnostics.AnalyzerExecutor
         .ExecuteAndCatchIfThrows_NoLock(DiagnosticAnalyzer analyzer, Action analyze, Nullable`1 info)'
```

## Root Cause (hypothesis)

The analyzer appears to use a LINQ `.First()` or `.Single()` call to look up the procedure parameter matching a `<param name="...">` doc tag. When the `name` attribute is absent (null or empty string), the lookup finds no matching parameter and throws instead of handling the case gracefully.

**Suggested fix:** Before the LINQ lookup, guard against a missing or empty `name` attribute and emit an appropriate diagnostic (e.g. "XML `<param>` tag is missing the `name` attribute") rather than calling `.First()` unconditionally.

## Environment

| Context | Version |
|---|---|
| ALCops version | 1.3.4 (stable channel) |
| VS Code extension | `arthurvdv.alcops` |
| AL Language extension | `ms-dynamics-smb.al` 17.0.2273547 |
| BC runtime | 17.0 |
| BC application | 28.0.0.0 |
| OS | Windows |

## Workaround

Remove all `<param></param>` tags that are missing the `name` attribute. Every `<param>` tag must have a `name` attribute that exactly matches one of the procedure's parameter names.

## Steps to Reproduce

Create a file `ReproduceCrash.Codeunit.al` in any AL project with `ALCops.DocumentationCop` enabled and build. The analyzer crashes on both patterns below.

**Pattern 1** — `<param>` without `name` attribute on a procedure that has parameters:

```al
codeunit 99999 ReproduceCrashPattern1
{
    /// <summary>Returns a greeting for the given name.</summary>
    /// <param></param>
    /// <returns>The greeting text.</returns>
    procedure Greet(Name: Text): Text
    begin
        exit('Hello, ' + Name);
    end;
}
```

**Pattern 2** — `<param>` without `name` attribute on a procedure that has **no parameters**:

```al
codeunit 99999 ReproduceCrashPattern2
{
    /// <summary>Returns the name of the current company.</summary>
    /// <param></param>
    /// <returns>The company name.</returns>
    procedure GetCompanyName(): Text[30]
    begin
        exit(CompanyName());
    end;
}
```

### File to change

```
src/ALCops.DocumentationCop/Analyzers/XmlDocumentationProcedureConsistency.cs
```

### Crash location (exact)

Inside `AnalyzeDocumentationComments`, the `case "param":` branch unconditionally calls `.First()`:

```csharp
// BEFORE — crashes when <param> has no name attribute
case "param":
    var nameAttribute = (XmlNameAttributeSyntax)element.StartTag.Attributes.First(att =>
        att.IsKind(EnumProvider.SyntaxKind.XmlNameAttribute));
    var parameterName = nameAttribute.Identifier.GetText().ToString();
    if (!docCommentParameters.ContainsKey(parameterName))
        docCommentParameters.Add(parameterName, element);
    else
        ctx.ReportDiagnostic(Diagnostic.Create(
            DiagnosticDescriptors.XmlDocumentationProcedureConsistency,
            element.GetLocation()));
    break;
```

### Fix

Replace `.First(...)` with `.FirstOrDefault(...)`, cast to nullable, and guard against `null` by reporting a diagnostic and breaking early:

```csharp
// AFTER — emits a diagnostic instead of crashing
case "param":
    var nameAttribute = element.StartTag.Attributes
        .FirstOrDefault(att => att.IsKind(EnumProvider.SyntaxKind.XmlNameAttribute))
        as XmlNameAttributeSyntax;
    if (nameAttribute is null)
    {
        // <param> tag is missing the required name attribute — report and skip
        ctx.ReportDiagnostic(Diagnostic.Create(
            DiagnosticDescriptors.XmlDocumentationProcedureConsistency,
            element.GetLocation()));
        break;
    }
    var parameterName = nameAttribute.Identifier.GetText().ToString();
    if (!docCommentParameters.ContainsKey(parameterName))
        docCommentParameters.Add(parameterName, element);
    else
        ctx.ReportDiagnostic(Diagnostic.Create(
            DiagnosticDescriptors.XmlDocumentationProcedureConsistency,
            element.GetLocation()));
    break;
```

### Test case to add

The existing test structure uses `.al` fixture files with inline `{|DC0005|}` markers. Add the following file:

**`src/ALCops.DocumentationCop.Test/Rules/XmlDocumentationProcedureConsistency/HasDiagnostic/ParamWithoutName.al`**

```al
codeunit 0 ParamWithoutNameTest
{
    /// <summary>Procedure with a param tag that has no name attribute.</summary>
    /// {|DC0005|}<param></param>
    /// <returns>Some value.</returns>
    procedure ProcedureWithParam(Value: Integer): Boolean
    begin
        exit(Value > 0);
    end;

    /// <summary>Parameterless procedure with a spurious param tag.</summary>
    /// {|DC0005|}<param></param>
    /// <returns>Company Name</returns>
    procedure ParameterlessProcedure(): Text[30]
    begin
        exit(CompanyName());
    end;
}
```

Then register the new test case in `XmlDocumentationProcedureConsistency.cs` (the test class):

```csharp
[TestCase("Return")]
[TestCase("Parameter")]
[TestCase("DuplicateParameter")]
[TestCase("DuplicateReturns")]
[TestCase("TryFunction")]
[TestCase("ParamWithoutName")]   // ← add this line
public async Task HasDiagnostic(string testCase)
```

### How to contribute

The project's README explicitly welcomes pull requests via the standard workflow:

1. **Fork** `https://github.com/ALCops/Analyzers`
2. **Create a branch** — e.g. `fix/dc0005-param-tag-missing-name-crashes-analyzer`
3. Apply the one-file fix to `XmlDocumentationProcedureConsistency.cs`
4. Add the `ParamWithoutName.al` test fixture and register the `[TestCase]`
5. **Open a pull request** against `main` referencing this issue
